### DISCUSSION TITLE
run_on_preflight option for ACL plugin

### DISCUSSION THREAD

**Discussion by DrPyser** [Kong/kong#7271] [Ideas and feature requests]
### Summary

https://discuss.konghq.com/t/acl-preflight-continue/351
The ACL plugin interferes with CORS preflight requests. If the ACL plugin is configured on a route, CORS preflight requests will be denied(since preflight requests won't be authenticated). CORS preflight requests should not be authenticated, as per the CORS standard.

### Steps To Reproduce

1. Configured & enable ACL plugin on a route
2. Send a CORS preflight request(OPTIONS) on that route(e.g. from a browser)

**Comment by alesanmed** [comment]
I'm having the exact sam problem which is blocking us from using the plugin. It would be really helpful to have this solved :(

**Comment by rmetcalf9** [comment]
I have this issue as well. (I don't know why it was moved from https://github.com/Kong/kong/issues/4423)

I am using the JWT plugin with the ACL Plugin
The JWT plugin will assign a consumer/group to the request as it looks up keys.
The ACL plugin then checks the correct consumer/group is assigned in order to allow correct access.

The problem is OPTIONS requests. The spec says that options request should not include credentials. This means there is no JWT to authorise and it is impossible to assign a consumer.

The JWT plugin has a nice option "run on preflight" which can be turned off. This means that it won't block the requests.
Unfortunately the ACL plugin will not have this option. As these requests have no consumer they are blocked :(

I want my downstream service to handle all options requests.

I can see this issue was raised 3 years ago and then it's been moved to a discussion.
I don't see any explanation why it was not implemented. Is there a logical reason why it has not been implemented?

I can work around this issue using the cors plugin but in my use case the downstream service is what is aware of what the options response should be and it is a pain to keep having to reflect the changes. I just want the options requests to not ever be blocked.

**Reply by rmetcalf9** [reply]
I have also worked out I could put the plugins on the route and not the service and then have two routes one for options with no plugins and one for the rest. It's a pain, having the preflight option would make my config a lot simpler.
